packages/toc/src/utils/markdown.ts

```diff
@@ -62,88 +62,41 @@ export async function getHeadingId(
   return null;
 }
 
-/**
- * Parses the provided string and returns a list of headings.
- *
- * @param text - Input text
- * @returns List of headings
- */
-export function getHeadings(text: string): IMarkdownHeading[] {
-  // Split the text into lines:
-  const lines = text.split('\n');
-
-  // Iterate over the lines to get the header level and text for each line:
-  const headings = new Array<IMarkdownHeading>();
-  let isCodeBlock;
-  let openingFence = 0;
-  let fenceType;
-  let lineIdx = 0;
-
-  // Don't check for Markdown headings if in a YAML frontmatter block.
-  // We can only start a frontmatter block on the first line of the file.
-  // At other positions in a markdown file, '---' represents a horizontal rule.
-  if (lines[lineIdx] === '---') {
-    // Search for another '---' and treat that as the end of the frontmatter.
-    // If we don't find one, treat the file as containing no frontmatter.
-    for (
-      let frontmatterEndLineIdx = lineIdx + 1;
-      frontmatterEndLineIdx < lines.length;
-      frontmatterEndLineIdx++
-    ) {
-      if (lines[frontmatterEndLineIdx] === '---') {
-        lineIdx = frontmatterEndLineIdx + 1;
-        break;
-      }
-    }
+export async function getHeadings(
```

Comment:
Can you rename this to `parseHeadings` and leave the old `getHeadings` as-is for backward compatibility, but mark the old `getHeadings` as deprecated in favour of async `parseHeadings`?